Sources/PenguinParallel/NonblockingThreadPool/NonBlockingThreadPool.swift

```diff
@@ -135,22 +135,22 @@ public class NonBlockingThreadPool<Environment: ConcurrencyPlatform>: ComputeThr
     perThreadKey.localValue = state
   }
 
-  public func dispatch(_ fn: @escaping Task) {
+  public func dispatch(_ fn: @escaping () -> Void) {
     if let local = perThreadKey.localValue {
       // Push onto local queue.
-      if let bounced = queues[local.threadId].pushFront(fn) {
+      if let bounced = queues[local.threadId].pushFront(.dispatch(fn)) {
         // If local queue is full, execute immediately.
-        bounced()
+        execute(bounced, localQueue: queues[local.threadId])
       } else {
         wakeupWorkerIfRequired()
       }
     } else {
       // Called not from within the threadpool; pick a victim thread from the pool at random.
       // TODO: use a faster RNG!
       let victim = Int.random(in: 0..<queues.count)
-      if let bounced = queues[victim].pushBack(fn) {
```

Comment:
Thought you were against `victim`.  It's not API, but still? Personally, I have no objections, but why waste an opportunity to give you a hard time?

Reply:
:-D